code/illustrating_side_constraint_issue.R

```diff
@@ -0,0 +1,264 @@
+library(mgcv)
```

Comment:
Should this file remain in the PR/repo? We don't use this anywhere now do we?

Reply:
Good point; I can delete it.